src/app/Fake.DotNet.FxCop/FxCop.fs

```diff
@@ -0,0 +1,212 @@
+      UseGAC : bool
+      /// Analyze only these types and members.  Default empty
+      Types : string seq
+      /// Update the project file if there are any changes.  Default false.
+      SaveResultsInProjectFile : bool
+      /// Working directory for relative file paths.  Default is the current working directory
+      WorkingDirectory : string
+      /// Give verbose output during analysis.  Default true.
+      Verbose : bool
+      /// The error level that will cause a build failure.  Default ontFailBuild.
+      FailOnError : ErrorLevel
+      /// Path to the FxCop executable.  Default = %VSINSTALLDIR%/Team Tools/Static Analysis Tools/FxCop/FxCopCmd.exe
+      /// where %VSINSTALLDIR% is a Visual Stdio 2017 installation location derived from the registry
+      ToolPath : string
+      /// Write output XML and project files even in the case where no violations
+      /// occurred.  Default false.
+      ForceOutput : bool
+      /// Custom dictionary used by spelling rules.  Default empty.
+      CustomDictionary : string }
+
+    static member private vsInstallPath() =
+        if Environment.isWindows then
+            let instance =
+                BlackFox.VsWhere.VsInstances.getWithPackage
+                    "Microsoft.VisualStudio.Component.Static.Analysis.Tools" false
+                |> List.tryHead
+            match instance with
+            | Some vs -> vs.InstallationPath
+            | None -> String.Empty
+        else String.Empty
+
+    /// FxCop Default parameters, values as above
+    static member Create() =
+        { ApplyOutXsl = false
+          DirectOutputToConsole = true
+          DependencyDirectories = Seq.empty
+          ImportFiles = Seq.empty
+          RuleLibraries = Seq.empty
+          Rules = Seq.empty
+          CustomRuleset = String.Empty
+          IgnoreGeneratedCode = false
+          ConsoleXslFileName = String.Empty
+          ReportFileName = Shell.pwd() @@ "FXCopResults.html"
+          OutputXslFileName = String.Empty
+          PlatformDirectory = String.Empty
+          ProjectFile = String.Empty
+          IncludeSummaryReport = true
+          Types = Seq.empty
+          UseGAC = false
+          SaveResultsInProjectFile = false
+          WorkingDirectory = Shell.pwd()
+          Verbose = true
+          FailOnError = ErrorLevel.DontFailBuild
+          ToolPath =
+              Params.vsInstallPath()
+              @@ "Team Tools/Static Analysis Tools/FxCop/FxCopCmd.exe"
+          ForceOutput = false
+          CustomDictionary = String.Empty }
+
+// default Xml reader
+let internal XmlReadIntBase failOnError xmlFileName nameSpace prefix xPath =
+  (Xml.read_Int failOnError xmlFileName nameSpace prefix xPath) |> snd
+
+// Unit test mocking point
+let mutable internal XmlReadInt = XmlReadIntBase
+
+/// This checks the result file with some XML queries for errors
+/// [omit]
+let checkForErrors resultFile =
```

Comment:
I guess this is public by design?

Reply:
If yes, why omit it from the docs?

Reply:
That one is just copied verbatim from [the legacy implementation](https://github.com/fsharp/FAKE/blob/release/next/src/legacy/FakeLib/FXCopHelper.fs#L52-L55), so the design question should really be addressed to the original author.

The function would extract results from FxCop output derived from other sources, so is useful to that extent, but I'm not invested one way or the other.